plugins/carla/CMakeLists.txt

```diff
@@ -0,0 +1,111 @@
+cmake_minimum_required(VERSION 3.16...3.25)
+
+legacy_check()
+
+option(ENABLE_CARLA "Enable building OBS with carla plugin host" ON)
+if(NOT ENABLE_CARLA)
+  target_disable(carla)
+  return()
+endif()
+
+# Find Qt
+find_qt(COMPONENTS Core Widgets)
+
+# Use pkg-config to find optional deps
+find_package(PkgConfig)
+
+# Find pthread via cmake
+if(OS_WINDOWS)
+  set(carla_pthread_libs OBS::w32-pthreads)
+else()
+  find_package(Threads REQUIRED)
+  set(carla_pthread_libs ${CMAKE_THREAD_LIBS_INIT})
+endif()
+
+# Optional: transient X11 window flags
+if(PKGCONFIG_FOUND AND NOT (OS_MACOS OR OS_WINDOWS))
+  pkg_check_modules(X11 "x11")
+else()
+  set(X11_FOUND FALSE)
+endif()
+
+# Use *.mm files under macOS, regular *.cpp everywhere else
+if(OS_MACOS)
+  set(CARLA_OBJCPP_EXT "mm")
+else()
+  set(CARLA_OBJCPP_EXT "cpp")
+endif()
+
+# Import extra carla libs
+include(cmake/jackbridge.cmake)
+add_library(carla::jackbridge ALIAS carla-jackbridge)
+
+include(cmake/lilv.cmake)
+add_library(carla::lilv ALIAS carla-lilv)
+
+include(cmake/rtmempool.cmake)
+add_library(carla::rtmempool ALIAS carla-rtmempool)
+
+include(cmake/water.cmake)
+add_library(carla::water ALIAS carla-water)
+
+# Setup binary tools
+include(cmake/carla-discovery-native.cmake)
+include(cmake/carla-bridge-native.cmake)
+
+# Setup carla-bridge target
+add_library(carla-bridge MODULE)
+add_library(OBS::carla-bridge ALIAS carla-bridge)
+
+target_compile_definitions(
+  carla-bridge
+  PRIVATE BUILDING_CARLA
```

Comment:
Far too many compiler definitions - are these all truly necessary? Also external value definitions should be avoided as much as possible because they can lead to unnecessary rebuilds.

Best practise is to create a single static library, excluded from the `ALL` target, with a generated header file that contains the variable data (using CMake's `configure_file` function), which then exposes the values via it's public symbols and a shared header file.

In that case the static "configuration" library contains all the variable values, which only requires this little library and linking it will make the actual target pick up all the values.

Reply:
something to keep in mind, wont be needed anymore but thanks